remotes/docker/fetcher.go

```diff
@@ -130,22 +128,14 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 			rc, err := r.open(ctx, req, desc.MediaType, offset)
 			if err != nil {
 				// Store the error for referencing later
-				if firstErr == nil {
-					firstErr = err
-				}
+				errorList = append(errorList, err)
 				continue // try another host
 			}
 
 			return rc, nil
 		}
 
-		if errdefs.IsNotFound(firstErr) {
-			firstErr = fmt.Errorf("could not fetch content descriptor %v (%v) from remote: %w",
-				desc.Digest, desc.MediaType, errdefs.ErrNotFound,
```

Comment:
The old code was returning an `errdefs.ErrNorFound`, which likely is important for code using this package.